### PR TITLE
Improve benchmarks

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

The number of documents to generate / insert is now separated from `-benchtime` / `b.N`.

## Readiness checklist

- [ ] I added/updated unit tests.
- [x] I added/updated integration/compatibility tests.
- [x] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/shareddata/benchmarks.go [hunk 1]
```diff
@@ -15,61 +15,68 @@
...
 
-	elements := make([]bson.E, n)
-	elements[0] = bson.E{"_id", 0}
+	var total int
 
-	for i := 1; i < len(elements); i++ {
-		elements[i] = bson.E{
-			Key:   genLetters(i, 50),
-			Value: values[i%vLen],
+	return func() bson.D {
```
Comment (chilagrow) [inline integration/shareddata/benchmarks.go:41]: I see, each time caller of this function calls it, it returns `values` from next index.

File: integration/shareddata/benchmarks.go [hunk 2]
```diff
@@ -79,57 +86,20 @@ var BenchmarkLargeDocuments = newGeneratorBenchmarkProvider("LargeDocuments", 12
 func AllBenchmarkProviders() []BenchmarkProvider {
 	providers := []BenchmarkProvider{
 		BenchmarkSmallDocuments,
-		BenchmarkLargeDocuments,
+		BenchmarkSettingsDocuments,
 	}
 
-	// check that names are unique and randomize order
+	// check that bse names are unique and randomize order
```
Comment (chilagrow) [inline integration/shareddata/benchmarks.go:92]:
```suggestion
	// check that base names are unique and randomize order
```

**Review by chilagrow — COMMENTED** [review]
Great to have more insert benchmark based on batch size! :hugs: 

I see data race in CI, could you have a look please?

File: integration/Taskfile.yml [hunk 3]
```diff
@@ -54,26 +56,63 @@ tasks:
     desc: "Run benchmarks for `pg` handler"
     cmds:
       - >
-        go test -run=XXX -count={{.COUNT}} -bench={{.BENCHNAME}} -benchtime={{.BENCHTIME}} -benchmem
+        go test -tags={{.BUILD_TAGS}} -timeout=3h -run=XXX
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply integration/Taskfile.yml]: Updated to use `-timeout=0` to make things more explicit

**Review by chilagrow — APPROVED** [review]
Looks great!

What I previously saw in CI data dace is no longer there either.

**Review by w84thesun — APPROVED** [review]
LGTM

**Pull request merged**